test/conformance/conformance_flowgraph.h

```diff
@@ -59,32 +62,523 @@ struct first_functor {
         return operator()(OutputType());
     }
 
-    void operator()( const OutputType& argument, oneapi::tbb::flow::multifunction_node<int, std::tuple<int>>::output_ports_type &op ) {
-        operator()(OutputType());
-        std::get<0>(op).try_put(argument);
+    void operator()( oneapi::tbb::flow::continue_msg, oneapi::tbb::flow::multifunction_node<oneapi::tbb::flow::continue_msg, std::tuple<oneapi::tbb::flow::continue_msg>>::output_ports_type &op ) {
+       operator()(OutputType());
+       std::get<0>(op).try_put(oneapi::tbb::flow::continue_msg());
     }
 };
 
 template<typename OutputType>
 std::atomic<int> first_functor<OutputType>::first_id;
 
+template< typename OutputType>
+struct counting_functor {
+    OutputType return_value;
 
-template< typename OutputType >
-struct inc_functor {
-    static std::atomic<size_t> execute_count;
+    static std::atomic<std::size_t> execute_count;
+
+    counting_functor(OutputType value = OutputType()) : return_value(value) {
+        execute_count = 0;
+    }
 
     OutputType operator()( oneapi::tbb::flow::continue_msg ) {
        ++execute_count;
-       return OutputType();
+       return return_value;
+    }
+
+    OutputType operator()( OutputType ) {
+       ++execute_count;
+       return return_value;
+    }
+
+    void operator()( const int&, oneapi::tbb::flow::multifunction_node<int, std::tuple<int>>::output_ports_type &op ) {
+       ++execute_count;
+       std::get<0>(op).try_put(return_value);
     }
 
-    OutputType operator()( int argument ) {
+    void operator()( oneapi::tbb::flow::continue_msg, oneapi::tbb::flow::multifunction_node<int, std::tuple<int>>::output_ports_type &op ) {
        ++execute_count;
-       return argument;
+       std::get<0>(op).try_put(return_value);
+    }
+
+    OutputType operator()( oneapi::tbb::flow_control & fc ) {
+       ++execute_count;
+       if(execute_count > std::size_t(return_value)){
+           fc.stop();
+           return return_value;
+       }
+       return return_value;
     }
 };
 
 template<typename OutputType>
-std::atomic<size_t> inc_functor<OutputType>::execute_count;
+std::atomic<std::size_t> counting_functor<OutputType>::execute_count;
+
+template< typename OutputType>
+struct worker_body {
+    OutputType operator()( oneapi::tbb::flow::continue_msg ) {
+       utils::doDummyWork(1000000);
+       return OutputType();
+    }
+    OutputType operator()( OutputType ) {
+        utils::doDummyWork(1000000);
+       return OutputType();
+    }
+    void operator()( const int& argument, oneapi::tbb::flow::multifunction_node<int, std::tuple<int>>::output_ports_type &op ) {
+        utils::doDummyWork(1000000);
+        std::get<0>(op).try_put(argument);
+    }
+};
+
+template< typename OutputType>
+struct dummy_functor {
+    OutputType operator()( oneapi::tbb::flow::continue_msg ) {
+       return OutputType();
+    }
+    OutputType operator()( OutputType ) {
+       return OutputType();
+    }
```

Comment:
```suggestion
    template<typename InputType>
    OutputType operator()( InputType ) {
       return OutputType();
    }
```

Such code reduction approach may be also applicable for other cases.